### PR TITLE
Fixes Issue #3272

### PR DESCRIPTION
**Pull request by sainiak009**
Screenshot attached:

[](https://postimg.org/image/emdssz06f/)

### CODE REVIEW

File: app/helpers/ticketing.py [hunk 1]
```diff
@@ -303,6 +303,9 @@ def initiate_order_payment(form, paid_via=None):
         if order:
             user = DataGetter.get_or_create_user_by_email(email, form)
             order.user_id = user.id
+            order.user.user_detail.firstname = form['firstname']
+            order.user.user_detail.lastname = form['lastname']
```
Comment (niranjan94) [inline app/helpers/ticketing.py]: If a name is already present in the fields do not overwrite it.

File: app/helpers/ticketing.py [hunk 2]
```diff
@@ -303,9 +303,9 @@ def initiate_order_payment(form, paid_via=None):
         if order:
             user = DataGetter.get_or_create_user_by_email(email, form)
             order.user_id = user.id
-            if not(order.user.user_detail.firstname or order.user.user_detail.lastname):
-                 order.user.user_detail.firstname = form['firstname']
-                 order.user.user_detail.lastname = form['lastname']
+            if not (order.user.user_detail.firstname or order.user.user_detail.lastname):
```
Comment (niranjan94) [inline app/helpers/ticketing.py]: No. in some cases, there maybe a firstname but no lastname. We don't want the name to get overwritten in those cases. 

Overwrite only if both firstname and lastname is not present in `user_detail`

**Comment by sainiak009** [comment]
@niranjan94 please check.

File: app/helpers/ticketing.py (hunk 2 above)
Comment (sainiak009) [inline app/helpers/ticketing.py]: Please review the changes.

File: app/helpers/ticketing.py [hunk 3]
```diff
@@ -303,6 +303,9 @@ def initiate_order_payment(form, paid_via=None):
         if order:
             user = DataGetter.get_or_create_user_by_email(email, form)
             order.user_id = user.id
+            if not (order.user.user_detail.firstname and order.user.user_detail.lastname):
```
Comment (niranjan94) [inline app/helpers/ticketing.py]: Logical error.. needs to be 
`if not order.user.user_detail.firstname and not order.user.user_detail.lastname`

Comment (sainiak009) [inline app/helpers/ticketing.py]: @niranjan94  Sorry for the silly mistake, Corrected it now.

**Comment by sainiak009** [comment]
@niranjan94 I have made the required changes in commit message. Now, Please review and merge this pull request.

**Comment by niranjan94** [comment]
@sainiak009 I think you mis-understood what I said about commit messages. Make sure your commit messages says what exactly was done in a commit. No need to put issue number in the commit. We have a PR for that.

See the below examples,

- `Fixes Issue #3272` - Doesn't say anything :x: 
- `Name missing in ticket email and invoice fixed #3272` - Says what was fixed. But not what change has been done in the commit :x: 
- `Store purchaser name in user detail record` - Tells the exact change that was done in that commit :white_check_mark:

**Comment by sainiak009** [comment]
@niranjan94 Yes, I misunderstood it. Corrected it now.

**Pull request merged**